app/commands/v2/put_link_set.rb

```diff
@@ -9,7 +9,29 @@ def call
           version.increment
           version.save! if link_set.valid?
 
-          link_set.links = merge_links(link_set.links, link_params.fetch(:links))
+          links_hash = link_params.fetch(:links)
+
+          links_hash.each do |link_type, target_content_ids|
+            if target_content_ids.empty?
+              Link.where(link_set: link_set, link_type: link_type).delete_all
+            else
+              old_target_ids = Link.where(link_type: link_type, link_set: link_set).map(&:target_content_id)
+
+              ids_to_be_deleted = old_target_ids - target_content_ids
+              ids_to_be_created = target_content_ids - old_target_ids
+
+              ids_to_be_deleted.each do |id|
```

Comment:
You can give active record queries arrays in a `where` clause, so this would do the same as these three lines, and would be one query:

```
link_set.links.where(link_type: link_type, target_content_id: ids_to_be_deleted).delete_all
```